.github/workflows/pull_request.yml

```diff
@@ -50,6 +67,18 @@ jobs:
         with:
           python-version: '3.12'
 
+      - name: Set up uv
+        run: curl -LsSf https://astral.sh/uv/install.sh | sh
```

Comment:
we could also move to using this: https://github.com/astral-sh/setup-uv

not a blocker though, since this works fine as well

Reply:
https://docs.astral.sh/uv/guides/integration/github/ has plenty of recommendations on how to do that, including caching.

Reply:
haha, I saw your suggestion just now. I figured it out independently. I have just pushed the change. Let's check the results...